File: Schedule/Schedule.Persistence/Entities/Lesson.cs
﻿namespace Schedule.Persistence.Entities;

public partial class Lesson
{
    public int LessonId { get; set; }

    public int Number { get; set; }

    public int? Subgroup { get; set; }

    public int? TimeId { get; set; }

    public int TimetableId { get; set; }

    public int? DisciplineId { get; set; }

    public bool IsChanged { get; set; }

    public virtual Discipline? Discipline { get; set; }

    public virtual ICollection<LessonTeacherClassroom> LessonTeacherClassrooms { get; } = new List<LessonTeacherClassroom>();

    public virtual Time? Time { get; set; }

    public virtual Template Timetable { get; set; } = null!;

    public virtual Timetable TimetableNavigation { get; set; } = null!;
}

File: api.schedule/schedule-main-service/internal/model/lesson.go
package model

import "time"

type Lesson struct {
	ID           int64
	Number       int64
	Start        time.Time
	End          time.Time
	Duration     time.Time
	TimetableID  int64
	DisciplineID int64
	TeacherIDs   []int64
	ClassroomIDs []int64
	IsChanged    bool
}
